kernel/build-kernel.sh

```diff
@@ -0,0 +1,252 @@
+#!
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+#---------------------------------------------------------------------
+# Description: This script is the *ONLY* to build a kernel for development.
+# Note: The default kernel version will be taked from:
+# https://github.com/kata-containers/runtime/blob/master/versions.yaml
+#---------------------------------------------------------------------
+
+# Display usage to stdout.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+readonly script_name="$(basename "${BASH_SOURCE[0]}")"
+readonly script_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+#flags and default values
+# GOPATH used by Go tools.
+GOPATH=${GOPATH:-${HOME}/go}
+# Directory where the kernel source is located
+kernel_path=""
+# Kernel version to be used
+kernel_version=""
+# Flag know if need to download the kernel source
+download_kernel=false
+# The repository where kernel configuration lives
+runtime_repository="github.com/kata-containers/runtime"
+# The repository where kernel configuration lives
+kernel_config_repo="github.com/kata-containers/packaging"
+# Defult path to search patches to apply to kernel
+readonly default_patches_dir="${GOPATH}/src/${kernel_config_repo}/kernel/patches/"
+# Defult path to search config for kata
+readonly default_kernel_config_dir="${GOPATH}/src/${kernel_config_repo}/kernel/configs/"
+# 
+patches_path=""
+#
+hypervisor_target=""
+#
+arch_target=""
+#
+kernel_config_path=""
+# destdir
+DESTDIR="${DESTDIR:-/}"
+#PREFIX=
+PREFIX="${PREFIX:-/usr}"
+#project_name
+readonly project_name="kata-containers"
+
+
+
+source "${script_dir}/../scripts/lib.sh"
+
+usage()
+{
+cat <<EOT
+Overview:
+
+	Build a kernel for Kata Containers
+
+Usage:
+
+	$script_name [options] <command> <argument>
+
+Commands:
+
+- setup 
+
+- build
+
+- install
+
+Options:
+
+	-c <path>: Path to config file to build a the kernel
+	-h       : Display this help.
+	-k <path>: Path to kernel to build
+	-p <path>: Path to a directory with patches to apply to kernel.
+	-v       : Kernel version to use if kernel path not provided.
+
+Example:
+	
+
+EOT
+}
+
+check_version_yaml()
+{
+	info "check versions.yaml"
+	local versions_yaml="${GOPATH}/src/${runtime_repository}/versions.yaml"
```

Comment:
Nit: you could define a variable for `versions.yaml` and reference it twice here.